server/src/main/java/org/opensearch/common/Rounding.java

```diff
@@ -451,9 +444,36 @@ protected Prepared maybeUseArray(long minUtcMillis, long maxUtcMillis, int max)
                 values = ArrayUtil.grow(values, i + 1);
                 values[i++] = rounded;
             }
-            return i <= LINEAR_SEARCH_ARRAY_ROUNDING_MAX_THRESHOLD
-                ? new BidirectionalLinearSearchArrayRounding(values, i, this)
-                : new BinarySearchArrayRounding(values, i, this);
+            return new ArrayRounding(values, i, this);
+        }
+    }
+
+    /**
+     * ArrayRounding is an implementation of {@link Prepared} which uses
+     * pre-calculated round-down points to speed up lookups.
+     */
+    private static class ArrayRounding implements Prepared {
+        private final Roundable roundable;
+        private final Prepared delegate;
+
+        public ArrayRounding(long[] values, int size, Prepared delegate) {
```

Comment:
I think this class should take the `Roundable` instead of `values` and `size` (it does not need these values but merely uses  `RoundableFactory` which caller could do:

```suggestion
        public ArrayRounding(Roundable roundable, Prepared delegate) {
```

Reply:
I like this. Done!